config/initializers/telegram_initializer.rb

```diff
@@ -2,11 +2,14 @@
 
 Rails.application.configure do
   config.after_initialize do
-    Telegram.bots_config = {
-      default: {
-        token: Setting.telegram_bot_api_key,
-        username: Setting.telegram_bot_username
-      }
-    }
+    config = {}
+    if ActiveRecord::Base.connection.table_exists? Organization.table_name
+      Organization.find_each do |org|
+        next unless org.telegram_configured?
+
+        config[org.id] = { token: org.telegram_bot_api_key, username: org.telegram_bot_username }
+      end
+    end
+    Telegram.bots_config = config
```

Comment:
I was using the `organization.name` as a key for the `Telegram.bots_config` first but then I remembered that `name` will be a mutable attribute. We would have to restart the server after a change. That's why I went for the `id` which is immutable and won't change.

Reply:
I'm not a big fan of promoting one Organization to `:default`. If `telegram-bot` depends on a single `:default` configuration so badly, that would be the moment where I would fix it on the gem's side.

Reply:
> I was using the `organization.name` as a key for the `Telegram.bots_config` first but then I remembered that `name` will be a mutable attribute. We would have to restart the server after a change. That's why I went for the `id` which is immutable and won't change.

this is what I mentioned to you about adding new organizations and it requiring a server restart. The biggest reason that I went with name over id is that id was not producing a valid route, which cannot start with a number. This was also an issue with the seeded organization, which is named 100eyes. I can see with your changes in https://github.com/tactilenews/100eyes/pull/1909/commits/76df1f34191a85e7d0a2d82dc7ba05a9311ee655, this should no longer be an issue.

Reply:
> I'm not a big fan of promoting one Organization to `:default`. If `telegram-bot` depends on a single `:default` configuration so badly, that would be the moment where I would fix it on the gem's side.
> 
> Is there any way to get around the `:default` bot?

Ok, I think I made some progress in my own understanding. Maybe you know this already, but I'll just write it out for our future selves. The big issue that I was running into was even start up the application with docker-compose since telegram-rb uses bot poller mode by default in dev env and that requires either having the env variable `BOT` that can be mapped to a key in the `Telegram.bots_config`, or it falls back to `:default`, which if not configured causes an issue. We could just have a `default` bot when in `Rails.env.development?`, but I don't really like that. I could also get it working by setting `BOT` in my `.env.development` file and/or in the `docker-compose.override.yml`, but it requires changing the `telegram_initializer` to convert the `org.id` to `org.id.to_s.to_sym`, which also needs to be matched in the telegram adapter.

I think the solution I liked the most was to remove the `telegram` service from the override altogether, which allows us to more closely mimic the webhook mode that we use in production. This requires running the `bot.set_webhook(url: 'https://development-100eyes.loca.lt/telegram/<bot_token_hash>')` command, but there is no need to delete the webhook for that bot to continue testing in the future.

It also gives me more confidence that not having a default bot configured should not affect us in production.